plugins/flatten/flatten_main.ml

```diff
@@ -0,0 +1,146 @@
+open Bap.Std
+open Core_kernel
+include Self()
+
+
+let get_direct_typ (e : exp) : Type.t = match e with
+  | Bil.Var v -> (match Var.typ v with
+    | Type.Imm n -> Type.Imm n
+    | Type.Mem (n, m) -> Type.Mem (n, m)
+    | Type.Unk -> Type.Unk)
```

Comment:
We can simplify the var case (probably you forgot to clean it after some debugging), and we forgot an important case of a flat expression - `Unknown`. It is also flat. 
```suggestion
  | Bil.Var v -> Var.typ v
  | Bil.Unknown (_,t) -> t
```